src/plugin/notification/helpers.ts

```diff
@@ -0,0 +1,36 @@
+import { uniq } from "lodash";
+import { CycleData } from "src/api-client/abstract_client";
+import { CycleReport } from "src/engine/interfaces";
+import { MUTEZ_FACTOR } from "src/utils/constants";
+import { sum } from "src/utils/math";
+import { NotificationInputData } from "./interfaces";
+
+export const getDataForPlugins = (
+  cycleData: CycleData,
+  cycleReport: CycleReport
+): NotificationInputData => {
+  const { cycle, delegatorPayments } = cycleReport;
+
+  return {
+    cycle: cycle.toString(),
+    cycleStakingBalance: cycleData.cycleStakingBalance.toString(),
+    totalDistributed: sum(...delegatorPayments.map((p) => p.amount))
+      .div(MUTEZ_FACTOR)
+      .dp(3)
+      .toString(),
+    numberOfDelegators: uniq(
+      delegatorPayments.map((x) => x.delegator)
+    ).length.toString(),
+  };
+};
+
+export const constructMessage = (
```

Comment:
Question here is - do we really want same format across platforms? We would be limited to plaintext without extras.

So I would move construct message to relevant plugins and decide there what is best message format.